### ISSUE TITLE
Change the fill_value from within Julia

### ISSUE THREAD

**Issue by felixcremer** [JuliaIO/Zarr.jl#39]
Is there a way to set the fill_value from within Julia. 

I can change the fill_value by manual setting it in the .zarray file inside my data folder, but it would be nicer to set it from within Julia. 

If I try to naively set it in the metadata type of my ZArray, I get the following error:
```julia
julia> zc2.metadata.fill_value = -99
ERROR: setfield! immutable struct of type Metadata cannot be changed
Stacktrace:
 [1] setproperty!(::Zarr.Metadata{Union{Missing, Float32},3,Zarr.BloscCompressor}, ::Symbol, ::Int64) at ./Base.jl:34
 [2] top-level scope at REPL[55]:1
 [3] eval(::Module, ::Any) at ./boot.jl:331
 [4] eval_user_input(::Any, ::REPL.REPLBackend) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.4/REPL/src/REPL.jl:86
 [5] run_backend(::REPL.REPLBackend) at /home/crem_fe/.julia/packages/Revise/XFtoQ/src/Revise.jl:1162
 [6] top-level scope at REPL[1]:0
```

**Comment by meggart** [comment]
I think the fill_value of a ZArray is something static and should not be changed on an existing array. What is wrong with setting the fillvalue during creation of the array?

**Comment by felixcremer** [comment]
I am not sure about the context of this anymore, but I think, that I tried to open a stack of Rasters via ArchGDAL and  saved them into Zarr. But they didn't pick up the correct fill_value so I tried to set the fill_value afterwards also to help with compression of the data, to get the nodata parts of the rasters compressed. 

With your hints, I am going to try to set the fill_value while saving the data into the zarr format.
